backend/tests/integration/pidstore/minters/test_minters_texkey.py

```diff
@@ -0,0 +1,169 @@
+#
+# Copyright (C) 2019 CERN.
+#
+# inspirehep is free software; you can redistribute it and/or modify it under
+# the terms of the MIT License; see LICENSE file for more details.
+import pytest
+from helpers.utils import create_record, override_config
+from invenio_pidstore.models import PersistentIdentifier, PIDStatus
+
+from inspirehep.pidstore.errors import PIDAlreadyExistsError
+
+
+def test_minter_mint_new_record_without_texkey(inspire_app):
+    data = {
+        "authors": [{"full_name": "Janeway, K."}],
+        "publication_info": [{"year": 2000}],
+    }
+    with override_config(FEATURE_FLAG_ENABLE_TEXKEY_MINTER=True):
+        record = create_record("lit", data=data)
+
+    expected_pid_value_starts_with = "Janeway:2000"
+    pid = (
+        PersistentIdentifier.query.filter(PersistentIdentifier.object_uuid == record.id)
+        .filter(PersistentIdentifier.pid_type == "texkey")
+        .one()
+    )
+    assert pid.pid_value.startswith(expected_pid_value_starts_with)
+
+
+def test_minter_mint_new_record_with_texkey(inspire_app):
+    data = {
+        "authors": [{"full_name": "Janeway, K."}],
+        "publication_info": [{"year": 2000}],
+        "texkeys": ["Janeway:2000abc"],
+    }
+    with override_config(FEATURE_FLAG_ENABLE_TEXKEY_MINTER=True):
+        record = create_record("lit", data=data)
+
+    expected_pid_value = "Janeway:2000abc"
+    pid = (
+        PersistentIdentifier.query.filter(PersistentIdentifier.object_uuid == record.id)
+        .filter(PersistentIdentifier.pid_type == "texkey")
+        .one()
+    )
+    assert pid.pid_value == expected_pid_value
+
+
+def test_minter_mint_new_record_with_texkey_when_texkey_already_in_use(inspire_app):
+    data = {
+        "authors": [{"full_name": "Janeway, K."}],
+        "publication_info": [{"year": 2000}],
+        "texkeys": ["Janeway:2000abc"],
+    }
+    with override_config(FEATURE_FLAG_ENABLE_TEXKEY_MINTER=True):
+        record = create_record("lit", data=data)
+
+        with pytest.raises(PIDAlreadyExistsError):
+            record = create_record("lit", data=data)
+
+
+def test_minter_mint_record_update_with_texkey_changed(inspire_app):
+    data = data = {
+        "authors": [{"full_name": "Janeway, K."}],
+        "publication_info": [{"year": 2000}],
+    }
+    with override_config(FEATURE_FLAG_ENABLE_TEXKEY_MINTER=True):
+        record = create_record("lit", data=data)
+        data = dict(record)
+        data["publication_info"][0]["year"] = 1999
+        record.update(data)
+
+    pid_active = PersistentIdentifier.query.filter(
+        PersistentIdentifier.pid_value == record["texkeys"][0]
+    ).one()
+    pid_old = PersistentIdentifier.query.filter(
+        PersistentIdentifier.pid_value == record["texkeys"][1]
+    ).one()
+
+    assert pid_old.pid_value != pid_active.pid_value
+
+    assert len(record["texkeys"]) == 2
+    assert pid_active.status == PIDStatus.REGISTERED
+    assert pid_old.status == PIDStatus.DELETED
```

Comment:
Why is it marked as deleted here? It still belongs to the record, but it's not the primary texkey.

Reply:
Yes, wrong requirements in task. After discussion IRL this behavior will be changed.